test/e2e/on-request-error/isr/isr.test.ts

```diff
@@ -0,0 +1,90 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry, waitFor } from 'next-test-utils'
+import { getOutputLogJson } from '../_testing/utils'
+
+const outputLogPath = 'output-log.json'
+
+const isPPREnabledByDefault = process.env.__NEXT_EXPERIMENTAL_PPR === 'true'
+
+describe('on-request-error - isr', () => {
+  const { next, skipped, isNextDev } = nextTestSetup({
+    files: __dirname,
+    skipDeployment: true,
+  })
+
+  if (skipped) {
+    return
+  }
+
+  if (isNextDev) {
+    it('should skip in development mode', () => {
+      // This ISR test is only applicable for production mode
+    })
+    return
+  }
+
+  async function matchRevalidateReason(
+    errorMessage: string,
+    revalidateReason: string
+  ) {
+    await retry(async () => {
+      const json = await getOutputLogJson(next, outputLogPath)
+      expect(json[errorMessage]).toMatchObject({
+        context: {
+          revalidateReason,
+        },
+      })
+    })
+  }
+
+  describe('app router ISR', () => {
+    // TODO: support isr reason in PPR for app page
+    if (!isPPREnabledByDefault) {
```

Comment:
I wonder if we should favor the PPR test manifest here rather than the inlining, just so it's easier for us to find skipped tests.